workflow/util/util.go

```diff
@@ -592,19 +594,37 @@ func IsWorkflowSuspended(wf *wfv1.Workflow) bool {
 	return false
 }
 
-// IsWorkflowTerminated returns whether or not a workflow is considered terminated
-func IsWorkflowTerminated(wf *wfv1.Workflow) bool {
-	if wf.Spec.ActiveDeadlineSeconds != nil && *wf.Spec.ActiveDeadlineSeconds == 0 {
-		return true
+// TerminateWorkflow terminates a workflow by setting its spec.shutdown to ShutdownStrategyTerminate
+func TerminateWorkflow(wfClient v1alpha1.WorkflowInterface, name string) error {
+	patchObj := map[string]interface{}{
+		"spec": map[string]interface{}{
+			"shutdown": wfv1.ShutdownStrategyTerminate,
+		},
 	}
-	return false
+	var err error
+	patch, err := json.Marshal(patchObj)
+	if err != nil {
+		return errors.InternalWrapError(err)
+	}
+	for attempt := 0; attempt < 10; attempt++ {
```

Comment:
Instead of loop,  you can use  ExponentialBackoff https://github.com/argoproj/argo/blob/b9b87b7fa0cd3177c2b89cacff189f4893c5af95/util/util.go#L32?

Reply:
https://godoc.org/k8s.io/apimachinery/pkg/util/wait

Reply:
Done!